src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -330,6 +330,51 @@ public static void WriteIndented()
             Assert.Contains(Environment.NewLine, json);
         }
 
+        [Fact]
```

Comment:
Please add tests validating that non-whitespace inputs throw.

Reply:
Waiting for a decision on the list of characters to support.

Reply:
Let's start with horizontal whitespace characters only, since `\r` and `\n` seem like a niche application to me. We can add them later if there's demand.

Reply:
Done.